gluten-core/src/main/scala/org/apache/spark/util/TaskResources.scala

```diff
@@ -245,10 +262,19 @@ class TaskResourceRegistry extends Logging {
       throw new IllegalArgumentException(
         String.format("TaskResource with ID %s is not registered", id))
     }
-    resources.get(id).asInstanceOf[T]
+    resources.get(id)._1.asInstanceOf[T]
   }
 
   private[util] def getSharedUsage(): SimpleMemoryUsageRecorder = synchronized {
     sharedUsage
   }
+
+  private[util] def removeResource(id: String): Unit = synchronized {
+    if (!resources.containsKey(id)) {
+      throw new IllegalArgumentException(
+        String.format("TaskResource with ID %s is not registered", id))
+    }
+    resourcesPriorityQueue.remove(resources.get(id))
```

Comment:
Looks to be that this is a O(n) call for `PriorityQueue#remove(...)`, I am not sure. Could you confirm? If it's uneasy to optimize, probably we can use a `TreeSet` instead.

Reply:
Thanks for advice. I agree that we use `TreeSet` here. However, an O(n) complexity shouldn't become a hotspot, as the value of n won't be large if we prevent caching lots of objects. If n is large, then the GC can be a more serious problem.

Reply:
Thank you for your explanation. It makes sense to me except that I think a structure like `PriorityQueue` should be used for specific reasons. If we'll introduce inefficient operations then a user may question us why we are using it. Would you agree on that?